widgetry/src/color.rs

```diff
@@ -25,6 +25,44 @@ impl fmt::Display for Color {
 pub enum FancyColor {
     RGBA(Color),
     LinearGradient(LinearGradient),
+
+    /// Once uploaded, textures are addressed by their id, starting from 1, from left to right, top
+    /// to bottom, like so:
+    ///
+    ///     ┌─┬─┬─┐
+    ///     │1│2│3│
+    ///     ├─┼─┼─┤
+    ///     │4│5│6│
+    ///     ├─┼─┼─┤
+    ///     │7│8│9│
+    ///     └─┴─┴─┘
+    ///
+    /// Texture(0) is reserved for a pure white (no-op) texture.
+    Texture(Texture),
+
+    /// The `color` parameter is multiplied by any color baked into the texture, so typically this
+    /// only makes sense for grayscale textures.
+    ColoredTexture(Color, Texture),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Texture(i32);
+
+#[allow(dead_code)]
+impl Texture {
+    pub const NOOP: Texture = Texture(0);
```

Comment:
just a guess at what might be useful - much of them are unused. I can omit them if you want to avoid dead code.